src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryService.kt

```diff
@@ -19,17 +19,30 @@ class TelemetryService(
   }
 
   fun trackGoalCreateEvent(goal: Goal) {
-    telemetryClient.trackEvent(GOAL_CREATE_EVENT, goal.goalCreateEventCustomDimensions())
-    with(Span.current()) {
-      goal.goalCreateEventCustomDimensions().forEach { (key, value) -> setAttribute(key, value) }
-      addEvent("$GOAL_CREATE_EVENT-via-span")
-    }
+    telemetryClient.trackEvent(
+      GOAL_CREATE_EVENT,
+      getExistingCustomDimensions().plus(createEventCustomDimensions(goal)),
```

Comment:
The idea is now back to using the `TelemetryClient` (which we know works), but getting the existing custom dimentions (username and clientId is what we are hoping for), and adding it ones we create for this event